addons/product/migrations/13.0.1.2/pre-migration.py

```diff
@@ -161,6 +161,33 @@ def insert_missing_product_template_attribute_value(env):
         )
 
 
+def merge_variants_with_same_attributes(env):
+    openupgrade.logged_query(
+        env.cr, """
+        WITH templates AS (
+            SELECT pt.id, pp.combination_indices, COUNT(*)
+            FROM product_product pp
+            JOIN product_template pt ON pt.id = pp.product_tmpl_id
+            WHERE pp.combination_indices IS NOT NULL
+            GROUP BY pt.id, pp.combination_indices
+            HAVING COUNT(*) > 1
+        )
+        SELECT pp.id, pp.product_tmpl_id
+        FROM product_product pp
+        JOIN product_template pt ON pt.id = pp.product_tmpl_id
+        JOIN templates ON (templates.id = pt.id
+            AND templates.combination_indices = pp.combination_indices)"""
+    )
+    templates = {}
+    for product_id, template_id in env.cr.fetchall():
+        templates.setdefault(template_id, []).append(product_id)
+    for template_id in templates:
+        openupgrade_merge_records.merge_records(
+            env, 'product.product', templates[template_id][1:],
+            templates[template_id][0], field_spec=None, method='sql',
```

Comment:
The question is now, should I put a field_spec?

Reply:
It depends if the standard fields require it or default merge values are enough

Reply:
Yeah, my best option is `{'default_code': 'first_not_null', 'active': 'or', 'barcode': 'first_not_null', 'volume': 'max', 'weight': 'max'}`. The problem I see is for custom modules that add float/char/boolean fields, which would need to add more things in this dict. Maybe we could add in openupgradelib a functionality like if the dict contains `'other_fields': 'preserve'`, then every other field is excluded and preserved.

Reply:
https://github.com/OCA/openupgradelib/pull/270